framework/src/modules/chain/block_processor_v0.js

```diff
@@ -0,0 +1,212 @@
+const { baseBlockSchema } = require('./blocks');
+
+const SIZE_INT32 = 4;
+const SIZE_INT64 = 8;
+
+const getBytes = block => {
+	const blockVersionBuffer = intToBuffer(
+		block.version,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const timestampBuffer = intToBuffer(
+		block.timestamp,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const previousBlockBuffer = block.previousBlock
+		? intToBuffer(block.previousBlock, SIZE_INT64, BIG_ENDIAN)
+		: Buffer.alloc(SIZE_INT64);
+
+	const numTransactionsBuffer = intToBuffer(
+		block.numberOfTransactions,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const totalAmountBuffer = intToBuffer(
+		block.totalAmount.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const totalFeeBuffer = intToBuffer(
+		block.totalFee.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const rewardBuffer = intToBuffer(
+		block.reward.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const payloadLengthBuffer = intToBuffer(
+		block.payloadLength,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const payloadHashBuffer = hexToBuffer(block.payloadHash);
+
+	const generatorPublicKeyBuffer = hexToBuffer(block.generatorPublicKey);
+
+	const blockSignatureBuffer = block.blockSignature
+		? hexToBuffer(block.blockSignature)
+		: Buffer.alloc(0);
+
+	return Buffer.concat([
+		blockVersionBuffer,
+		timestampBuffer,
+		previousBlockBuffer,
+		numTransactionsBuffer,
+		totalAmountBuffer,
+		totalFeeBuffer,
+		rewardBuffer,
+		payloadLengthBuffer,
+		payloadHashBuffer,
+		generatorPublicKeyBuffer,
+		blockSignatureBuffer,
+	]);
+};
+
+const validateSchema = ({ block }) => {
+	const errors = validator.validate(baseBlockSchema, block);
+	if (errors.length) {
+		throw errors;
+	}
+};
+
+class BlockProcessorV0 extends BlockProcessor {
+	constructor({ blocksModule, logger, constants, exceptions }) {
+		super();
+		this.blocksModule = blocksModule;
+		this.logger = logger;
+		this.constants = constants;
+		this.exceptions = exceptions;
+
+		this.init.pipe([() => this.blocksModule.init()]);
+
+		this.validate.pipe([
+			data => this._validateVersion(data),
+			validateSchema,
+			({ block }) => getBytes(block),
+			(data, blockBytes) =>
+				this.blocksModule.validate({
+					...data,
+					blockBytes,
+				}), // validate common block header
+		]);
+
+		this.fork.pipe([
+			data => this.blocksModule.forkChoice(data), // validate common block header
+		]);
+
+		// TODO: fix this by adding back the validateNew function
+		this.validateNew.pipe([() => Promise.resolve()]);
+
+		this.verify.pipe([data => this.blocksModule.verify(data)]);
+
+		this.apply.pipe([data => this.blocksModule.apply(data)]);
+
+		this.applyGenesis.pipe([data => this.blocksModule.applyGenesis(data)]);
+
+		this.undo.pipe([data => this.blocksModule.undo(data)]);
+
+		this.create.pipe([data => this._create(data)]);
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	get version() {
+		return 0;
+	}
+
+	async _create({ transactions, previousBlock, keypair, timestamp }) {
```

Comment:
Why its a private method?

Reply:
because `_create` is registered to the `create` pipeline